pkg/setting/setting_smtp.go

```diff
@@ -38,4 +47,31 @@ func (cfg *Cfg) readSmtpSettings() {
 	cfg.Smtp.SendWelcomeEmailOnSignUp = emails.Key("welcome_email_on_sign_up").MustBool(false)
 	cfg.Smtp.TemplatesPatterns = util.SplitString(emails.Key("templates_pattern").MustString("emails/*.html, emails/*.txt"))
 	cfg.Smtp.ContentTypes = util.SplitString(emails.Key("content_types").MustString("text/html"))
+
+	// populate static headers
+	if err := cfg.readGrafanaSmtpStaticHeaders(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validHeader(header string) bool {
+	return smtpHeaderRegex.MatchString(header)
+}
+
+func (cfg *Cfg) readGrafanaSmtpStaticHeaders() error {
+	staticHeadersSection := cfg.Raw.Section("smtp.static_headers")
+	keys := staticHeadersSection.Keys()
+	cfg.Smtp.StaticHeaders = make(map[string]string, len(keys))
+
+	for _, key := range keys {
+		if !validHeader(key.Name()) {
+			return fmt.Errorf("invalid label name %q in [smtp.static_headers] configuration", key.Name())
```

Comment:
I think the error message could provide a better hint to how to fix the issue:

```suggestion
			return fmt.Errorf("header %q in [smtp.static_headers] configuration: must follow canonical MIME form", key.Name())
```